src/NCalc/LambdaExpressionVistor.cs

```diff
@@ -159,7 +159,17 @@ public override void Visit(Function function)
             switch (function.Identifier.Name.ToLowerInvariant())
             {
                 case "if":
-                    _result = L.Expression.Condition(args[0], args[1], args[2]);
+                    var if_arg1 = args[1];

```

Comment:
how about something like this, I think this would be more generic:

```csharp
var numberTypePriority = new Type[] { typeof(double), typeof(float), typeof(long), typeof(int), typeof(short) };
var index1 = numberTypePriority.IndexOf(args[1]);
var index2 = numberTypePriority.IndexOf(args[2]);
if (index1 >=0 && index2 >=0 && index1 != index2)
{
    args[1] = L.Expression.Convert(args[1], numberTypePriority[Math.Min(index1, index2));
    args[2] = L.Expression.Convert(args[2], numberTypePriority[Math.Min(index1, index2));
}

_result = L.Expression.Condition(args[0], args[1], args[2]);
```

Reply:
Updated. Thanks you, i appreciate your help.